src/light.cpp

```diff
@@ -29,33 +29,55 @@ static u8 light_LUT[LIGHT_SUN + 1];
 // The const ref to light_LUT is what is actually used in the code
 const u8 *light_decode_table = light_LUT;
 
+struct LightingParams {
+	float a, b, c; // polynomial coefficients
+	float boost, center, sigma; // normal boost parameters
+	float gamma;
+};
+
+static LightingParams params;
+
+float decode_light_f(float x)
+{
+	x = rangelim(x, 0.0f, 1.0f);
```

Comment:
Since this is called with argument 1.0 half the time, the first action should be to return 1.0 in that situation.